frontend/packages/knative-plugin/src/components/add/event-sources/EventSourcesSelector.tsx

```diff
@@ -26,6 +25,21 @@ const EventSourcesSelector: React.FC<EventSourcesSelectorProps> = ({ eventSource
     setErrors,
     setStatus,
   } = useFormikContext<FormikValues>();
+
+  const [recommended, setRecommended] = React.useState(false);
+  const recommendApplicationName = React.useCallback(() => {
+    setFieldValue('application.name', EVENT_SOURCES_APP);
+    setFieldTouched('application.name', true);
+  }, [setFieldTouched, setFieldValue]);
```

Comment:
nit: do we need it as used only in one place now and can be used inside `useEffect`